README.md

```diff
@@ -1,17 +1,83 @@
 # Core Data Modules
-
 [![CircleCI](https://circleci.com/gh/AfricasVoices/experimental_CoreDataModules/tree/master.svg?style=shield)](https://circleci.com/gh/AfricasVoices/experimental_CoreDataModules/tree/master)
 
-The skeleton of an AVF module library, with tests. 
-Currently contains a trivial example which cleans gender from English text.
+A library of reusable AVF functionality and data structures.
+
+It includes the primary data structure in which all runs are stored, helper functions 
+for IO and hashing, and the cleaning library.
+
+## Installation
+CoreDataModules is not available on PyPI, so must instead be installed from this GitHub repository.
+Do so with the following command, setting `<version>` to the tag to install e.g. `v0.0.1`:
+```
+$ pipenv install -e git+ssh://git@github.com/AfricasVoices/CoreDataModules.git@<version>#egg=CoreDataModules
+```
+
+## API Overview
+
+#### TracedData
+The data format for a batch of runs in the data pipeline is always an `iterable of TracedData`.
+
+A `TracedData` object can be thought of as a "dictionary with data provenance".
+It provides the same read interface as a Python `dict`, but writes must be performed using the `append_data` method,
+which requires some additional metadata about each update when it is called. `TracedData` objects maintain a complete
+record of all previous updates.
+
+To construct a `TracedData` object, read a value, and update value(s):
+```python
+import time
+from core_data_modules.traced_data import TracedData, Metadata
+
+USER = "test_user"
+
+td = TracedData({"age": "20", "language": "english"}, Metadata(USER, Metadata.get_call_location(), time.time()))
+print(td["age"])  # 20
+
+td.append_data({"age": "21"}, Metadata(USER, Metadata.get_call_location(), time.time()))
+print(td["age"])  # 21
+```
+
+#### IO
+CoreDataModules supports conversion between in-memory `iterables of TracedData` and Coda, CSV, and JSON files.
+For example, to save and load a list of `TracedData` objects by serializing to and from JSON:
+```python
+from core_data_modules.traced_data.io import TracedDataJsonIO
+
+data =  # TODO: assign iterable of TracedData objects
+
+with open("output.json", "w") as f:
+    TracedDataJsonIO.export_traced_data_iterable_to_json(data, f)
+    
+with open("input.json", "r") as f:
+    data = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
+```
+
+#### Cleaning
```

Comment:
This is a layering violation - this structure knows nothing about cleaning - could we move this to the demonstrator repo readme?

Reply:
I don't understand - the cleaners in this repository. Not showing how to use them here would be like not showing how to use TracedData.

Reply:
Ahh, I see.

In which case, I think we should make just make the IO a subsection of TracedData (all of the IO functions use trace data IIRC?).

Reply:
That makes more sense. Updated as requested.